scarlet/component.py

```diff
@@ -356,31 +340,52 @@ def __init__(self, frame, cube, bbox=None):
 
     @property
     def cube(self):
-        return self._pad_cube(self._parameters[0]._data)
+        return self._parameters[0]._data
 
     def get_model(self, *parameters):
         cube = None
         for p in parameters:
             if p._value is self._parameters[0]:
-                cube = self._pad_cube(p)
+                cube = p
 
         if cube is None:
             cube = self.cube
 
         return cube
 
-    def _pad_cube(self, cube):
-        if self.bbox is not None:
-            padded = np.pad(cube, self.pad_width, mode="constant", constant_values=0)
-            return padded[self.slices]
-        return cube
+
+@primitive
+def _add_models(*models, full_model, slices):
+    """Insert the models into the full model
+    """
+    for i in range(len(models)):
+        full_model[slices[i][0]] += models[i][slices[i][1]]
+    return full_model
+
+
+def _grad_add_models(upstream_grad, *models, full_model, slices, index):
+    """Gradient for a single model
+
+    The full model is just the sum of the models,
+    so the gradient is 1 for each model,
+    we just have to slice it appropriately.
+    """
+    model = models[index]
+    full_model_slices = slices[index][0]
+    model_slices = slices[index][1]
+
+    def result(upstream_grad):
+        _result = np.zeros(model.shape, dtype=model.dtype)
+        _result[model_slices] = upstream_grad[full_model_slices]
+        return _result
+    return result
 
 
 class ComponentTree:
     """Base class for hierarchical collections of Components.
     """
 
-    def __init__(self, components):
+    def __init__(self, components, trim=True):
```

Comment:
Why is the `trim` option needed? It cuts the model to the intersection of all component boxes, but what's the purpose of doing so?

Reply:
This is the part I'm most unsure about. The single `Component` models now return a model that is _not_ projected into the full model but instead is the size of each source. I feel like `ComponentTree` should do the same thing, except if it's a `Blend` and we want to get the full model to compare with the observation. So this was my first thought about an easy way to do this, but I'll spend some time over the weekend thinking about a better way. If you have one I'm happy to hear it.

Reply:
If `trim==True`, `ComponentTree.get_model` returns a cube without telling the downstream customer the proper box for it. That can't be right